test/e2e/networking.go

```diff
@@ -29,6 +29,60 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+var _ = BeforeSuite(func() {
+	//Assert basic external connectivity.
+	//Since this is not really a test of kubernetes in any way, we
+	//leave it as a pre-test assertion, rather than a Ginko test.
+	resp, err := http.Get("http://google.com")
+	if err != nil {
+		Failf("Unable to connect/talk to the internet: %v", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		Failf("Unexpected error code, expected 200, got, %v (%v)", resp.StatusCode, resp)
+	}
+})
+
+func LaunchNetTestPodPerNode(nodes *api.NodeList, name string, c *client.Client, ns string) []string {
+
+	podNames := []string{}
+
+	totalPods := len(nodes.Items)
+
+	Expect(totalPods).NotTo(Equal(0))
+
+	for i, node := range nodes.Items {
+		podName := fmt.Sprintf("%s-%d", name, i)
+		podNames = append(podNames, podName)
+		Logf("Creating pod %s on node %s", podName, node.Name)
+		_, err := c.Pods(ns).Create(&api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name: podName,
+				Labels: map[string]string{
+					"name": name,
+				},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "webserver",
+						Image: "gcr.io/google_containers/nettest:1.1",
+						Args: []string{
+							"-service=" + name,
+							//peers >= totalPods should be asserted by the container.
+							fmt.Sprintf("-peers=%d", totalPods),
```

Comment:
This identifies the number of peers including itself. Much of this code has already been reviewed so it's a shame that much of it is turning up in green again.